### PR TITLE
Release build api

### PR DESCRIPTION
**Pull request by fractalwrench**

### CODE REVIEW

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagPluginExtension.groovy [hunk 1]
```diff
@@ -4,13 +4,28 @@ package com.bugsnag.android.gradle
...
-    def boolean autoUpload = true
-    def boolean autoProguardConfig = true
-    def boolean uploadDebugBuildMappings = false
-    def boolean overwrite = false
-    def int retryCount = 0
-    def boolean ndk = false
-    def String sharedObjectPath = null
+
+    String endpoint = 'https://upload.bugsnag.com'
+    String releasesEndpoint = 'https://releases.example.com' // TODO change me!
```
Comment (fractalwrench) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagPluginExtension.groovy]: Will need to update this to the production API Endpoint

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagReleasesTask.groovy [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    private static final String MK_GRADLE_VERSION = "gradle.version"
+    private static final String MK_GIT_VERSION = "git.version"
+
+    BugsnagReleasesTask() {
+        super()
+        this.description = "Assembles information about the build that will be sent to the releases API"
+    }
+
+    @TaskAction
+    void fetchReleaseInfo() { // TODO handle manual task
```
Comment (fractalwrench) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagReleasesTask.groovy]: Need to check that this task can be called manually

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagPluginExtension.groovy [hunk 3]
```diff
@@ -4,13 +4,28 @@ package com.bugsnag.android.gradle
...
-    def boolean autoUpload = true
-    def boolean autoProguardConfig = true
-    def boolean uploadDebugBuildMappings = false
-    def boolean overwrite = false
-    def int retryCount = 0
-    def Boolean ndk = null
-    def String sharedObjectPath = null
+
+    String endpoint = 'https://upload.bugsnag.com'
+    String releasesEndpoint = 'https://releases.example.com' // TODO change me!
```
Comment (kattrali) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagPluginExtension.groovy]: ready to be real 😃

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagReleasesTask.groovy [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            releaseStage = project.bugsnag.releaseStage
+        }
+
+        root.put("appVersion", versionName)
+        root.put("appVersionCode", versionCode)
+        root.put("releaseStage", releaseStage)
+        root.put("autoAssignRelease", project.bugsnag.autoAssignRelease)
+
+        if (project.bugsnag.builderName != null) {
+            root.put("builderName", project.bugsnag.builderName)
```
Comment (snmaynard) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagReleasesTask.groovy]: Can we default to whoami?

File: src/main/groovy/com/bugsnag/android/gradle/BugsnagReleasesTask.groovy [hunk 5]
```diff
@@ -106,6 +106,9 @@ class BugsnagReleasesTask extends BugsnagVariantOutputTask {
 
         if (project.bugsnag.builderName != null) {
             root.put("builderName", project.bugsnag.builderName)
+        } else {
+            root.put("builderName", runCmd("whoami"))
+            project.logger.lifecycle("User: " + runCmd("whoami"))
```
Comment (kattrali) [inline src/main/groovy/com/bugsnag/android/gradle/BugsnagReleasesTask.groovy]: The debug logging should use a cached value rather than rerunning `whoami`

**Pull request merged**